libs/utils/src/json_stream.ts

```diff
@@ -0,0 +1,78 @@
+function isIterable(value: unknown): value is Iterable<unknown> {
+  return (
+    typeof value === 'object' && value !== null && Symbol.iterator in value
+  );
+}
+
+/**
+ * Stream a JSON-serializable value as a series of strings. In addition to the
+ * standard JSON types, this also supports `Iterable` values such as
+ * `Generator`.
+ *
+ * @example
+ *
+ * ```ts
+ * function* generateLargeArray() {
+ *   for (let i = 0; i < 1_000_000_000; i++) {
+ *     yield i;
+ *   }
+ * }
+ *
+ * const stream = jsonStream({ foo: [1, 2, 3], bar: generateLargeArray() });
+ * const writer = createWriteStream('output.json');
+ * for (const chunk of stream) {
+ *  writer.write(chunk);
+ * }
+ * ```
+ */
+export function* jsonStream(input: unknown, indent = 0): Generator<string> {
+  if (input === null) {
+    yield 'null';
+  } else if (
+    typeof input === 'boolean' ||
+    typeof input === 'number' ||
+    typeof input === 'string'
+  ) {
+    yield JSON.stringify(input);
+  } else if (isIterable(input)) {
+    const indentString = ' '.repeat(indent * 2);
+    const nextIndent = indent + 1;
+    const nextIndentString = ' '.repeat(nextIndent * 2);
+    yield '[';
+    let hasEntries = false;
+    for (const value of input) {
+      if (hasEntries) {
+        yield ',';
+      } else {
+        hasEntries = true;
+      }
+      yield `\n${nextIndentString}`;
```

Comment:
If `indent` is 0, should we skip line breaks?

Reply:
I guess I was just thinking if the caller specifies no indent, maybe they would want a compressed version with no line breaks. Similar to:

```js
> JSON.stringify([1, 2], null, 2)
'[\n  1,\n  2\n]'
> JSON.stringify([1, 2], null, 0)
'[1,2]'
```

Reply:
Ah, I see. Yeah that makes sense. I added a `compact` option for that and it's `true` by default.